content/blog/eng/a-redux-pattern.md

```diff
@@ -0,0 +1,203 @@
+---
+authors: ["pierroberto-lucisano"]
+comments: true
+date: "2022-04-21"
+draft: true
+share: true
+categories: [Design patterns, Javascript]
+title: "A Redux pattern"
+languageCode: "en-EN"
+type: "post"
+toc: true
+twitterImage: /images/a-redux-pattern/elevator.jpg
+---
+
+# Introduction
+
+There is inconsistency around the Redux community about how to use actions. Redux Toolkit documentation suggests the following approach:
+
+> [...] we recommend trying to treat actions more as "describing events that occurred", rather than "setters".
+
+Why should we treat actions as events rather than setters? Dan Abramov, founder of Redux, said that Redux doesn't reinvent event sourcing. It's up to people how to use it. It's clear that there isn't a well accepted approach about how to use actions.
+
+In this article I'll talk about a design pattern for Redux. I'll show you how it can help resolve some known problems. In particular, I'll try to find a way to describe how to treat actions. This we'll lead us to deal better with the asynchronicity and the mutation problems. The main goal is to get a consistent state for our application. To achieve that, we'll look for a predictable way to mutate the state of the application.
```

Comment:
I think, from your premises, that this article presupposes certain knowledge. So why not to explicit the prerequisites?